test/test_out_s3.rb

```diff
@@ -456,6 +464,33 @@ def test_write_with_custom_s3_object_key_format_containing_hex_random_placeholde
     FileUtils.rm_f(s3_local_file_path)
   end
 
+  def test_write_with_zstd
+    setup_mocks(true)
+    s3_local_file_path = "/tmp/s3-test.zst"
+  
+    expected_s3path = "log/events/ts=20110102-13/events_0-#{Socket.gethostname}.zst"
+  
+    setup_s3_object_mocks(s3_local_file_path: s3_local_file_path, s3path: expected_s3path)
+  
+    config = CONFIG_TIME_SLICE + "\nstore_as zstd\n"
+    d = create_time_sliced_driver(config)
+  
+    time = event_time("2011-01-02 13:14:15 UTC")
+    d.run(default_tag: "test") do
+      d.feed(time, { "a" => 1 })
+      d.feed(time, { "a" => 2 })
+    end
+  
+    File.open(s3_local_file_path, 'rb') do |file|
+      compressed_data = file.read
+      uncompressed_data = Zstd.decompress(compressed_data)
+      expected_data = %[2011-01-02T13:14:15Z\ttest\t{"a":1}\n] +
+                      %[2011-01-02T13:14:15Z\ttest\t{"a":2}\n]
+      assert_equal expected_data, uncompressed_data
+    end
+    FileUtils.rm_f(s3_local_file_path)
+  end
```

Comment:
Let's remove unnecessary spaces.

```suggestion
  def test_write_with_zstd
    setup_mocks(true)
    s3_local_file_path = "/tmp/s3-test.zst"

    expected_s3path = "log/events/ts=20110102-13/events_0-#{Socket.gethostname}.zst"

    setup_s3_object_mocks(s3_local_file_path: s3_local_file_path, s3path: expected_s3path)

    config = CONFIG_TIME_SLICE + "\nstore_as zstd\n"
    d = create_time_sliced_driver(config)

    time = event_time("2011-01-02 13:14:15 UTC")
    d.run(default_tag: "test") do
      d.feed(time, { "a" => 1 })
      d.feed(time, { "a" => 2 })
    end

    File.open(s3_local_file_path, 'rb') do |file|
      compressed_data = file.read
      uncompressed_data = Zstd.decompress(compressed_data)
      expected_data = %[2011-01-02T13:14:15Z\ttest\t{"a":1}\n] +
                      %[2011-01-02T13:14:15Z\ttest\t{"a":2}\n]
      assert_equal expected_data, uncompressed_data
    end
    FileUtils.rm_f(s3_local_file_path)
  end
```